core/embed/rust/src/ui/model_mercury/component/hold_to_confirm.rs

```diff
@@ -0,0 +1,310 @@
+            0.0,
+            easer::functions::Cubic,
+            0.2,
+            pareen::constant(1.0),
+        );
+
+        let header_opacity = pareen::constant(0.0).seq_ease_out(
+            0.1,
+            easer::functions::Cubic,
+            0.3,
+            pareen::constant(1.0),
+        );
+
+        let header_opacity2 = pareen::constant(1.0).seq_ease_in(
+            2.0,
+            easer::functions::Cubic,
+            0.2,
+            pareen::constant(0.0),
+        );
+
+        let circle_opacity = pareen::constant(1.0).seq_ease_out(
+            0.1,
+            easer::functions::Cubic,
+            0.1,
+            pareen::constant(0.0),
+        );
+
+        let pad_color = pareen::constant(0.0).seq_ease_in_out(
+            0.1,
+            easer::functions::Cubic,
+            1.9,
+            pareen::constant(1.0),
+        );
+
+        let ciclem_max_height = pareen::constant(0.0).seq_ease_in(
+            0.1,
+            easer::functions::Cubic,
+            1.5,
+            pareen::constant(1.0),
+        );
+
+        let final_circle_radius = pareen::constant(0.0).seq_ease_in(
+            1.6,
+            easer::functions::Cubic,
+            0.6,
+            pareen::constant(1.0),
+        );
+
+        let haptic = pareen::constant(0.0).seq_ease_in(
+            0.0,
+            easer::functions::Linear,
+            Self::DURATION,
+            pareen::constant(1.0),
+        );
+
+        let t = self.timer.elapsed().into();
+
+        let o1 = parent_cover_opacity.eval(t);
+        let o2 = header_opacity.eval(t);
+        let o3 = circle_opacity.eval(t);
+        let o4 = header_opacity2.eval(t);
+
+        let c1 = Color::lerp(
+            theme::GREY_EXTRA_DARK,
+            Color::rgb(0x09, 0x74, 0x48),
+            pad_color.eval(t),
+        );
+
+        let h1 = ciclem_max_height.eval(t);
+        let r1 = final_circle_radius.eval(t);
+
+        let hap = haptic.eval(t);
+
+        (o1, o2, o3, o4, c1, h1, r1, hap)
+    }
+
+    pub fn start(&mut self) {
+        self.timer.start();
+    }
+
+    pub fn reset(&mut self) {
+        self.timer = Stopwatch::new_stopped();
+    }
+}
+
+/// Component requesting an action from a user. Most typically embedded as a
+/// content of a Frame and promptin "Tap to confirm" or "Hold to XYZ".
+#[derive(Clone)]
+pub struct HoldToConfirm {
+    title: Label<'static>,
+    area: Rect,
+    button: Button,
+    circle_color: Color,
+    circle_pad_color: Color,
+    circle_inner_color: Color,
+    anim: HoldToConfirmAnim,
+}
+
+#[derive(Clone)]
+enum DismissType {
+    Tap,
+    Hold,
+}
+
+impl HoldToConfirm {
+    pub fn new() -> Self {
+        let button = Button::new(ButtonContent::Empty)
+            .styled(theme::button_default())
+            .with_long_press(Duration::from_millis(2200))
+            .without_haptics();
+        Self {
+            title: Label::new(
+                "Continue\nHolding".into(),
```

Comment:
Should we use `TR`. I'd suggest `instructions__continue_holding`.

I remember we discussed a problem with the newline in other langs though.

Reply:
https://github.com/trezor/trezor-firmware/pull/3776/commits/834ede0d3f37cd4d9d10e1d5a89626cef78385b6

went with one string and "\n" usage in translation file